mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -163,25 +187,28 @@ class StringIndexerModel (
     }
     transformSchema(dataset.schema, logging = true)
 
+    val metadata = NominalAttribute.defaultAttr
+      .withName($(outputCol)).withValues(labels).toMetadata()
+    // If we are skipping invalid records, filter them out.
+    val (filteredDataset, keepInvalid) = getHandleInvalid match {
```

Comment:
minor style comment: instead of keepInvalid, do you think that indexInvalid might be a better name (?)